bench/db_bench.ml

```diff
@@ -99,35 +107,47 @@ module Index = struct
         l "\tread amplification in bytes = %f; in nb of reads = %f " ratio_bytes
           ratio_reads)
 
-  let init () =
-    if Sys.file_exists root then (
-      let cmd = Printf.sprintf "rm -rf %s" root in
-      Logs.app (fun l -> l "exec: %s\n" cmd);
-      let _ = Sys.command cmd in
-      () )
+  let cleanup () =
+    let clean_dir dir =
+      let files = [ "data"; "log"; "lock"; "log_async"; "merge" ] in
+      List.iter
+        (fun file ->
+          let file = dir // file in
+          if Sys.file_exists file then (
+            let cmd = Printf.sprintf "rm %s" file in
+            Logs.app (fun l -> l "exec: %s\n" cmd);
+            let _ = Sys.command cmd in
+            () ))
+        files
+    in
+    for i = 0 to 4 do
```

Comment:
```suggestion
    for i = 0 to 4 do
```
Suspicious magic number 4. Does this come from the fact that you know `names` is called 5 times?

Reply:
yes :) I know it's not great, I don't see a nice way to refactor this (as cleanup is called at the beginning of a run).
I can instead have a list of all possible index names used in the benchmarks, maybe that is cleaner?

Reply:
This is perhaps over-complicating things, but we could keep the benchmark names outside in the `bench_list` variable, then have each benchmark consume a constructor namespaced with that string:
```ocaml
let namespace_v bench_name ~fresh ~log_size ~readonly name =
  Index.v ~fresh ~log_size ~readonly (root // bench_name // name)

...

let match_input ~bench ~triggers ~bench_name =
  if List.mem input triggers then
    let () = Log.app (fun l -> l "\n %s" bench_name) in
    bench (namespace_v ~bench_name)
```

This way you can clean by iterating over `bench_list` and purging all of the `root // bench_name` directories.

Reply:
Thanks ! I changed it as you proposed, and I think it is better.